plugin/plaintextmodel.cpp

```diff
@@ -264,6 +277,46 @@ bool PlainTextModel::readLines(
     return false;
 }
 
+QTextCodec *PlainTextModel::detectCodec(const char *encodedCharacters, qint64 length)
+{
+    QLocale locale;
+    QTextCodec *codec = QTextCodec::codecForLocale();
+
+    UErrorCode error = U_ZERO_ERROR;
+    if (UCharsetDetector * const detector = ucsdet_open(&error)) {
+        ucsdet_setText(detector, encodedCharacters, length, &error);
+
+        int32_t count = 0;
+        if (U_FAILURE(error)) {
+            qWarning() << "Unable to detect text encoding" << u_errorName(error);
+        } else if (const UCharsetMatch ** const matches = ucsdet_detectAll(detector, &count, &error)) {
+            QTextCodec *bestMatch = nullptr;
+
+            for (int32_t i = 0; i < count; ++i) {
+                if (QTextCodec *detectedCodec = QTextCodec::codecForName(ucsdet_getName(matches[i], &error))) {
+                    if (detectedCodec->name() == codec->name() || locale.name().left(locale.name().indexOf("_")) == QString(ucsdet_getLanguage(matches[i], &error))) {
+                        bestMatch = detectedCodec;
```

Comment:
As detected codecs are ordered as best ones first, this will in effect choose the worst one that matches language? Or if there's a language based codec with high confidence, this will override it with the default utf-8 if found later in the array.

Maybe could get better results if just giving language based detection some confidence boost (say, multiply with 1.5 or something), and picking the highest resulting value?

Could also split that long lines.

Reply:
I've done it. Please, take a look at it.

Reply:
> Maybe could get better results if just giving language based detection some confidence boost (say, multiply with 1.5 or something), and picking the highest resulting value?
> 

What do you mean?